src/NumericalAlgorithms/LinearSolver/ExplicitInverse.hpp

```diff
@@ -0,0 +1,246 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DenseMatrix.hpp"
+#include "DataStructures/DenseVector.hpp"
+#include "NumericalAlgorithms/Convergence/HasConverged.hpp"
+#include "NumericalAlgorithms/LinearSolver/LinearSolver.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/CreateIsCallable.hpp"
+
+namespace LinearSolver::Serial {
+
+/// \cond
+template <typename LinearSolverRegistrars>
+struct ExplicitInverse;
+/// \endcond
+
+namespace Registrars {
+/// Registers the `LinearSolver::Serial::ExplicitInverse` linear solver
+using ExplicitInverse = Registration::Registrar<Serial::ExplicitInverse>;
+}  // namespace Registrars
+
+namespace detail {
+CREATE_IS_CALLABLE(reset)
+CREATE_IS_CALLABLE_V(reset)
+}  // namespace detail
+
+/*!
+ * \brief Linear solver that builds a matrix representation of the linear
+ * operator and inverts it directly
+ *
+ * This solver first constructs an explicit matrix representation by "sniffing
+ * out" the operator, i.e. feeding it with unit vectors, and then directly
+ * inverts the matrix. The result is an operator that solves the linear problem
+ * in a single step. This means that each element has a large initialization
+ * cost, but all successive solves converge immediately.
+ *
+ * \par Advice on using this linear solver:
+ *
+ * - This solver is entirely agnostic to the structure of the linear operator.
+ *   Be advised to implement a linear solver that is specialized for your linear
+ *   operator to take advantage of its properties, e.g. its tensor-product
+ *   structure. Only use this solver if no alternatives are available and if you
```

Comment:
OK, I understand now.  I think this will read much better if you change "Be advised" to "It is advisable" or "It is usually better" or something like on of those.